File: proc_speed_test.py
#!/usr/bin/env python3

from __future__ import print_function
from glob import glob
import sys


n=0
e=0

for j in range(10000):
	for g in glob("/proc/[0-9]*/stat"):
		try:
			with open(g,"r") as f:
				s=f.readline()
				n+=1
		except Exception:
			e+=1

print("Count: ",n)
print("Error: ",e)

File: proc_speed_test.pl
#!/usr/bin/env perl

my $file;
my $line;
my $count;
my $err;

open( $file, "</proc/self/stat" ) or die "Hiba";
$line = readline($file);
close($file);



for( my $i=0; $i < 10000; $i++ ){
	foreach $filename (glob("/proc/[0-9]*/stat")) {
		if( open($file, "<", $filename) ){
			$line=readline($file);
			close($file);
			$count++;
		} else {
			$err++;
		}
	}
}

printf("Count: %u\n", $count);
printf("Error: %u\n", $err);

